engine/common/synchronization/engine.go

```diff
@@ -291,10 +291,14 @@ func (e *Engine) onSyncResponse(originID flow.Identifier, res *messages.SyncResp
 
 // onBlockResponse processes a response containing a specifically requested block.
 func (e *Engine) onBlockResponse(originID flow.Identifier, res *messages.BlockResponse) {
-	e.log.Debug().Str("origin_id", originID.String()).Msg("received block response")
 	// process the blocks one by one
+	min := res.Blocks[0].Header.Height
```

Comment:
Better to check if res.Blocks has at least one block, otherwise it's vulnerable to attack.

If there is no block at all, we could simply drop the message

Reply:
Great spot once again!